Proton/Sources/Attachment/Attachment.swift

```diff
@@ -304,51 +304,54 @@ open class Attachment: NSTextAttachment, BoundsObserving {
     ///   - position: Position in the text container.
     ///   - charIndex: Character index
     public override func attachmentBounds(for textContainer: NSTextContainer?, proposedLineFragment lineFrag: CGRect, glyphPosition position: CGPoint, characterIndex charIndex: Int) -> CGRect {
-         guard let textContainer = textContainer, textContainer.size.height > 0 && textContainer.size.width > 0 else { return .zero }
+        guard let textContainer = textContainer, textContainer.size.height > 0 && textContainer.size.width > 0 else { return .zero }
+
+        let indent: CGFloat
+        if charIndex < containerEditorView?.contentLength ?? 0 {
+            let paraStyle = containerEditorView?.attributedText.attribute(.paragraphStyle, at: charIndex, effectiveRange: nil) as? NSParagraphStyle
+            indent = paraStyle?.firstLineHeadIndent ?? 0
+        } else {
+            indent = 0
+        }
+        // Account for text leading and trailing margins within the textContainer
+        let adjustedContainerSize = CGSize(
+            width: textContainer.size.width - textContainer.lineFragmentPadding * 2 - indent,
+            height: textContainer.size.height
+        )
+        let adjustedLineFrag = CGRect(
+            x: lineFrag.origin.x,
+            y: lineFrag.origin.y,
+            width: min(lineFrag.size.width, adjustedContainerSize.width),
+            height: lineFrag.height
+        )
 
         var size: CGSize
 
         if let boundsProviding = contentView as? DynamicBoundsProviding {
-            size = boundsProviding.sizeFor(containerSize: textContainer.size, lineRect: lineFrag)
+            size = boundsProviding.sizeFor(containerSize: adjustedContainerSize, lineRect: adjustedLineFrag)
         } else {
             size = contentView?.bounds.integral.size ?? view.bounds.integral.size
         }
 
-        if size == .zero,
-            let fittingSize = contentView?.systemLayoutSizeFitting(textContainer.size) {
+        if (size.width == 0 || size.height == 0),
+            let fittingSize = contentView?.systemLayoutSizeFitting(adjustedContainerSize) {
             size = fittingSize
         }
 
         switch self.size {
         case .matchContent:
             size = contentView?.bounds.integral.size ?? view.bounds.integral.size
         case let .fixed(width):
-            size = CGSize(width: min(size.width, width), height: size.height)
+            size.width = min(size.width, width)
```

Comment:
I am nor sure why I wasn't using `size.width = width` instead of using `min`. In all cases, I believe the value for both will always be the same 🤔 I'll likely revise this in future.